file.go

```diff
@@ -47,13 +47,19 @@ func OpenFile(r io.ReaderAt, size int64, options ...FileOption) (*File, error) {
 		return nil, err
 	}
 
+	if cast, ok := f.reader.(interface{ SetMagicHeaderSection(offset, length int64) }); ok {
+		cast.SetMagicHeaderSection(0, 4)
+	}
 	if _, err := r.ReadAt(b[:4], 0); err != nil {
 		return nil, fmt.Errorf("reading magic header of parquet file: %w", err)
 	}
 	if string(b[:4]) != "PAR1" {
 		return nil, fmt.Errorf("invalid magic header of parquet file: %q", b[:4])
 	}
 
+	if cast, ok := f.reader.(interface{ SetMagicFooterSection(offset, length int64) }); ok {
+		cast.SetMagicFooterSection(size-8, 8)
+	}
```

Comment:
> Similarly here, maybe we don't need this one since the underlying reader could assume this without reading any of the content?

Hmm, there's an implicit assumption here that the application using the reader has knowledge of the size of the parquet file - not sure if we should require that, I'd prefer to include the offset size in this call.

> Alternatively, since this always part of the footer, we could make this part of the SetFooterSection call?

This depends on how we manage the cached objects. Should the caching layer intelligent-enough to detect when the request for a small object with a given offset & length can be satisfied with a larger stored object with offset & length that includes the smaller one?

Reply:
I feel like I don't have the answer here, you're closer to the problem than I am so you have a better sense of what model will serve you best. 

I'm happy to go with your current submission! We can always revisit later on when we've learned more about how well the current model works.

Reply:
Thanks! Yeah I think this model ensures that the external application/caching layer can rely on the library to pass on information about offsets and filesize, etc. I believe it will work well for us, but as you said, I'll definitely stay on top of this in case the design doesn't suit other users, etc.